src/otaclient/client_package.py

```diff
@@ -129,7 +135,7 @@ def _prepare_client_package(
 
         # ------ step 2: download the target package ------ #
         _package_filename = _available_package_metadata.filename
-        _package_file = cfg.OTACLIENT_INSTALLATION_RELEASE + "/" + _package_filename
+        _package_file = str(self._package_install_dir) + "/" + _package_filename
```

Comment:
Not recommended to use `+` operator for string operations, also we should not use string operation against Path joining. Please use `str(self._package_install_dir / _package_filename)` instead.

Reply:
Thank you! In the code, `str` and `Path` are mixture for `StrOrPath`.
Fix to use `Path` in `__init__` in https://github.com/tier4/ota-client/pull/547/commits/fffa901e7e28d5680f9adc6e515a96bb8c61bdf4.